### PR TITLE
T05.01

### PR DESCRIPTION
**Pull request by Innborne**
-restructuring of project components
-changed names of components and directories
-added card header and body components
-added button to delete selected cards

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -60,15 +60,23 @@ function App() {
...
+
   return (
     <div>
-      <CardsHeader>{header}</CardsHeader>
-      <CardsMain items={cards} OnChangeCards={changeCardsHandler}></CardsMain>
+      <Header>{header}</Header>
+      <CardList
+        items={cards}
+        OnChangeCards={changeCardsHandler}
+        OnDeleteCards={deleteCardsHandler}
```
Comment (kkaroukinIBA) [inline src/App.js]: Can you please refactor it passing only one method?

File: src/components/Card/Card.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        headerTitle={userInput.enteredTitle}
+        onTitleChange={titleChangeHandler}
+        onCheck={card.check}
+        editMode={editMode}
+        readOnly={readOnly}
+        onSaveClick={onSave}
+        onEditClick={onEdit}
+        onCancelClick={onCancel}
+        handleClick={onHandle}
+      ></CardHeader>
```
Comment (kkaroukinIBA) [inline src/components/Card/Card.js]: <Component {...props} />

File: src/components/Card/Card.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      <CardBody
+        bodyText={userInput.enteredText}
+        editMode={editMode}
+        onTextChange={textChangeHandler}
+      ></CardBody>
```
Comment (kkaroukinIBA) [inline src/components/Card/Card.js]: the same

File: src/components/Card/CardBody.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import "./CardBody.css";
+
+function CardBody(props) {
+  const cardTextEditMode = () => {
+    return (
+      <textarea
+        className="card-body-edit"
+        value={props.bodyText}
+        onChange={(event) => props.onTextChange(event.target.value)}
+      ></textarea>
```
Comment (kkaroukinIBA) [inline src/components/Card/CardBody.js]: the same

File: src/components/Card/CardBody.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    );
+  };
+
+  const cardTextReadMode = () => {
+    return <p className="card-body">{props.bodyText}</p>;
+  };
+  return <>{props.editMode ? cardTextEditMode() : cardTextReadMode()}</>;
```
Comment (kkaroukinIBA) [inline src/components/Card/CardBody.js]: Can you please refactor it by removing <> and </> from this line?

File: src/components/Card/CardHeader.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+function CardHeader(props) {
+  const cardTitleEditMode = () => {
+    return (
+      <div className="card-header-container">
+        <input
+          type="text"
+          className="card-header-edit-title"
+          value={props.headerTitle}
+          onChange={(event) => props.onTitleChange(event.target.value)}
+        ></input>
```
Comment (kkaroukinIBA) [inline src/components/Card/CardHeader.js]: the same

File: src/components/Card/CardHeader.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        <div className="card-header-edit-buttons">
+          <FiSave onClick={props.onSaveClick}></FiSave>
+          <FiX onClick={props.onCancelClick}></FiX>
```
Comment (kkaroukinIBA) [inline src/components/Card/CardHeader.js]: the same

File: src/components/Card/CardHeader.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+      <div className="card-header-container">
+        <h2 className="card-header-title">{props.headerTitle}</h2>
+        <div className="card-header-edit-buttons">
+          {!props.readOnly && <FiEdit3 onClick={props.onEditClick}></FiEdit3>}
+          <input
+            className="card-header-checkbox"
+            type="checkbox"
+            checked={props.onCheck}
+            onChange={props.handleClick}
+          ></input>
```
Comment (kkaroukinIBA) [inline src/components/Card/CardHeader.js]: the same

File: src/components/CardList/CardList.js [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+          {readOnlyCheckboxText}
+        </CardListCheckbox>
+      </div>
+      {props.items.map((cardData) => (
+        <Card
+          onSaveCardData={saveCardDataHandler}
+          key={cardData.id}
+          card={cardData}
+          readOnly={readOnlyMode}
+        ></Card>
```
Comment (kkaroukinIBA) [inline src/components/CardList/CardList.js]: the same

**Review by kkaroukinIBA — CHANGES_REQUESTED** [review]

File: src/App.js [hunk 10]
```diff
@@ -57,18 +57,14 @@ function App() {
...
+  const changeCardsHandler = (newCards) => {
+    setCards(newCards);
   };
 
   return (
     <div>
-      <CardsHeader>{header}</CardsHeader>
-      <CardsMain items={cards} OnChangeCards={changeCardsHandler}></CardsMain>
+      <Header>{header}</Header>
+      <CardList items={cards} OnChangeCards={changeCardsHandler}></CardList>
```
Comment (kkaroukinIBA) [inline src/App.js:67]: EMPTY TAG!!!!!111!!!1!

**Review by kkaroukinIBA — APPROVED** [review]

File: src/App.js (hunk 10 above)
Comment (Innborne) [reply src/App.js:67]: Will be fixed in next task

**Pull request merged**